projects/jupyter-server-ydoc/jupyter_server_ydoc/app.py

```diff
@@ -193,13 +197,54 @@ async def get_document(
 
         elif path is not None or content_type is not None or file_format is not None:
             raise ValueError(error_msg)
-        else:
-            room_id = room_id
+
+        if not self.ywebsocket_server.started.is_set():
+            asyncio.create_task(self.ywebsocket_server.start())
+            await self.ywebsocket_server.started.wait()
 
         try:
             room = await self.ywebsocket_server.get_room(room_id)
         except RoomNotFound:
-            return None
+            if not create:
```

Comment:
I worry about race conditions. I think that the logic may need to be wrapped in a `room_lock` as in:

https://github.com/jupyterlab/jupyter-collaboration/blob/7a3986a5be99e258497cc4b51a7755824137b107/projects/jupyter-server-ydoc/jupyter_server_ydoc/handlers.py#L69-L135

For that to work we need to create the `room_locks` dictionary outside of the `YDocWebSocketHandler` and pass it to both the handler and use it here.